pkg/operator/encryption/controllers/condition_controller.go

```diff
@@ -0,0 +1,214 @@
+// fully migrated read-key in the current config, and no later key is of identity type.
+type conditionController struct {
+	operatorClient operatorv1helpers.OperatorClient
+
+	queue         workqueue.RateLimitingInterface
+	eventRecorder events.Recorder
+
+	preRunCachesSynced []cache.InformerSynced
+
+	encryptedGRs []schema.GroupResource
+
+	encryptionSecretSelector metav1.ListOptions
+
+	deployer     statemachine.Deployer
+	secretClient corev1client.SecretsGetter
+}
+
+func NewConditionController(
+	deployer statemachine.Deployer,
+	operatorClient operatorv1helpers.OperatorClient,
+	kubeInformersForNamespaces operatorv1helpers.KubeInformersForNamespaces,
+	secretClient corev1client.SecretsGetter,
+	encryptionSecretSelector metav1.ListOptions,
+	eventRecorder events.Recorder,
+	encryptedGRs []schema.GroupResource,
+) *conditionController {
+	c := &conditionController{
+		operatorClient: operatorClient,
+
+		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "EncryptionConditionController"),
+		eventRecorder: eventRecorder.WithComponentSuffix("encryption-condition-controller"),
+
+		encryptedGRs: encryptedGRs,
+
+		encryptionSecretSelector: encryptionSecretSelector,
+		deployer:                 deployer,
+		secretClient:             secretClient,
+	}
+
+	c.preRunCachesSynced = setUpInformers(deployer, operatorClient, kubeInformersForNamespaces, c.eventHandler())
+
+	return c
+}
+
+func (c *conditionController) sync() error {
+	if ready, err := shouldRunEncryptionController(c.operatorClient); err != nil || !ready {
+		return err // we will get re-kicked when the operator status updates
+	}
+
+	currentConfig, desiredState, _, transitioningReason, err := statemachine.GetEncryptionConfigAndState(c.deployer, c.secretClient, c.encryptionSecretSelector, c.encryptedGRs)
+	if err != nil || len(transitioningReason) > 0 {
+		return err
+	}
+
+	cond := operatorv1.OperatorCondition{
+		Type:    "Encrypted",
+		Status:  operatorv1.ConditionTrue,
+		Reason:  "EncryptionComplete",
+		Message: fmt.Sprintf("Resources all encrypted: %s", grString(c.encryptedGRs)),
+	}
+	currentState := encryptionconfig.ToEncryptionState(currentConfig)
+
+	// check for identity key in desired state
+	foundIdentity := false
+	for _, s := range desiredState {
+		if s.WriteKey.Mode == state.Identity {
+			foundIdentity = true
+			break
+		}
+	}
+	if foundIdentity {
+		cond.Status = operatorv1.ConditionFalse
+		cond.Reason = "EncryptionDisabled"
+		cond.Message = "Encryption mode set to identity"
+	} else {
+
+		// now that the desired state look like it won't lead to identity as write-key, test the current state
+	NextResource:
+		for _, gr := range c.encryptedGRs {
+			foundFullyMigratedReadKey := false
+
+			s, ok := currentState[gr]
+			if ok {
+				if s.WriteKey.Mode == state.Identity {
+					cond.Status = operatorv1.ConditionFalse
+					cond.Reason = "EncryptionDisabled"
```

Comment:
do we want to distinguish `Encryption Disabled` vs `EncryptionIsBeingDisabled`?

Reply:
I wanted to avoid that kind of detail.